### PR TITLE
Issue #13345: Enable Package Annotation Check Examples Test

### PR DESCRIPTION
**Pull request by piyush-1234**
Issue: #13345

### CODE REVIEW

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java [hunk 1]
```diff
@@ -6,7 +6,8 @@
 </module>
 */
 
+package com.puppycrawl.tools.checkstyle.checks.annotation.packageannotation;
+
 // xdoc section -- start
-@Deprecated
-package com.example.annotations.packageannotation; //violation
+class Example1 {}
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java:12]: Please put deprecated annotation over package .
You can put `// xdoc section -- start` above it too`

**Review by romani — CHANGES_REQUESTED** [review]
Apply to all

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java (hunk 1 above)
Comment (piyush-1234) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java:12]: hi @romani , looks like, we have already anotate **package com.puppycrawl.tools.checkstyle.checks.annotation.packageannotation** inside package-info.java.

And again we are anotating the same package within **Example1.class** which is usually not possible as In Java, annotations should typically be placed in a special file named **package-info.java**.

And one more thing as we know about @Deprecated anotation is, In Java, the @Deprecated annotation is typically used to mark classes, methods, or fields that are considered outdated or should not be used anymore. It is not meant to be used above package imports in a Java class.

compilations error getting while doing so.
Screenshot 2023-11-08 213900
Screenshot 2023-11-08 213832

@romani 
need you input on this

Comment (piyush-1234) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java:12]: @romani any though on this?

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java:12]: Try to make move example to another package/folder
Like: com.puppycrawl.tools.checkstyle.checks.annotation.packageannotationexample1

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java [hunk 2]
```diff
@@ -6,6 +6,7 @@
 </module>
 */
 
+@Deprecated
 package com.puppycrawl.tools.checkstyle.checks.annotation.packageannotation;
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java:10]: I pushed extra commit to see if this compilation problem from javac or IDEA compiler

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java:10]: ok I see compile error:
```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/rivanov/java/github/romani/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java:[9,1] package annotations should be in file package-info.java
[ERROR] /home/rivanov/java/github/romani/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/package-info.java:[7,1] package com.puppycrawl.tools.checkstyle.checks.annotation.packageannotation has already been annotated
[INFO] 2 errors 

```

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/packageannotation/Example1.java:10]: https://checkstyle.org/checks/annotation/packageannotation.html#Description

> For Java SE8 and above, placement of package annotations in the package-info.java file is enforced by the compiler and this check is not necessary.

ok we need to put file in non-compileable folder of examples.

**Comment by romani** [comment]
I pushed update that we expect.

@piyush-1234 , please choose another migration Check for now, this PR update was very complicated for new contributors, so I did all updated for you.

**Review by romani — APPROVED** [review]
ok to merge if CI pass

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
GitHub, generate website

**Comment by piyush-1234** [comment]
> I pushed update that we expect.
> 
> @piyush-1234 , please choose another migration Check for now, this PR update was very complicated for new contributors, so I did all updated for you.

sure @romani

**Review by nrmancuso — APPROVED** [review]
TIL you could annotate package declarations in java < 8.

**Pull request merged**